app/src/main/java/ru/prometeydev/movie/ui/MovieNotifications.kt

```diff
@@ -0,0 +1,143 @@
+package ru.prometeydev.movie.ui
+
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.graphics.drawable.Drawable
+import android.util.Log
+import android.widget.RemoteViews
+import androidx.annotation.IdRes
+import androidx.annotation.WorkerThread
+import androidx.core.app.*
+import androidx.core.app.NotificationManagerCompat.IMPORTANCE_HIGH
+import androidx.core.graphics.drawable.toBitmap
+import androidx.core.net.toUri
+import coil.ImageLoader
+import coil.annotation.ExperimentalCoilApi
+import coil.request.Disposable
+import coil.request.ImageRequest
+import coil.target.Target
+import ru.prometeydev.movie.R
+import ru.prometeydev.movie.model.domain.Movie
+import java.lang.Exception
+
+class MovieNotifications(private val context: Context) {
+
+    private val imageLoader = ImageLoader.Builder(context)
+        .availableMemoryPercentage(0.25)
+        .crossfade(true)
+        .build()
+
+    private val notificationManagerCompat = NotificationManagerCompat.from(context)
+
+    fun initialize() {
+        if (notificationManagerCompat.getNotificationChannel(NOTIFY_CHANNEL_ID) == null) {
+            notificationManagerCompat.createNotificationChannel(
+                NotificationChannelCompat.Builder(NOTIFY_CHANNEL_ID, IMPORTANCE_HIGH)
+                    .setName(context.getString(R.string.chanel_new_top_rated_movie))
+                    .setDescription(context.getString(R.string.chanel_new_top_rated_movie_description))
+                    .build()
+            )
+        }
+    }
+
+    @ExperimentalCoilApi
+    @WorkerThread
+    suspend fun show(movie: Movie) {
+        val contentUri = "https://movie.prometeydev.ru/movie/${movie.id}".toUri()
+
+        val smallContentView =
+            RemoteViews(context.packageName, R.layout.notification_small).apply {
+                setTextViewText(
+                    R.id.appears,
+                    context.getString(R.string.chanel_new_top_rated_movie_description)
+                )
+                setTextViewText(R.id.movie_name, movie.title)
+                setTextViewText(
+                    R.id.rating,
+                    context.getString(R.string.rating_text, movie.ratings.toString())
+                )
+            }
+
+        val bigContentView = RemoteViews(context.packageName, R.layout.notification_big).apply {
+            setTextViewText(
```

Comment:
if you want to avoid boilerplate with find view by id start using view binding.

Reply:
Maybe next time, it will be necessary to read, ViewBinding has not yet used, there is only experience with DataBinding.